qiskit/util.py

```diff
@@ -111,6 +111,24 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def deprecate_function(msg):
+    """Emit a warning prior to calling decorated function.
+
+    Args:
+        msg (str): Warning message to emit.
+
+    Returns:
+        Callable: The decorated, deprecated callable.
+    """
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            warnings.warn(msg, DeprecationWarning, stacklevel=2)
+            return func(*args, **kwargs)
+        return wrapper
+    return decorator
```

Comment:
Wouldn't this also work?
```
def deprecate_function(msg, func):
    warnings.warn(msg, DeprecationWarning, stacklevel=2)
    return func
````
Seems overly complicated.

Reply:
What you have above would warn on importing of the module. This only warns when the function is called and also preserves the function docstrings and signatures.

Reply:
is it possible to make this be called only once? some function could be called hundreds times and it may be annoying.

Reply:
Fixed in 0752e2140.